crates/primitives/src/alias.rs

```diff
@@ -55,8 +51,34 @@ impl From<Alias> for String {
     }
 }
 
-impl From<&Alias> for String {
-    fn from(alias: &Alias) -> Self {
-        alias.0.to_owned()
+impl TryFrom<String> for Alias {
+    type Error = InvalidAlias;
+
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        Self::new(value).ok_or(InvalidAlias(
+            "alias exceeds maximum length of 50 characters",
+        ))
     }
 }
+
+impl<'de> Deserialize<'de> for Alias {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let encoded = String::deserialize(deserializer)?;
+        encoded.parse().map_err(|_| {
```

Comment:
`parse` takes `&str` which will force a reallocation, you should be calling `.try_into()` instead